core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,363 @@
+                "exactly %s bytes, which means the default value should be of exactly 2 * %s length hex string",
+            fixedString, expectedLength, expectedLength);
+        generator.writeString(fixedString);
+        break;
+      case BINARY:
+        Preconditions.checkArgument(
+            defaultValue instanceof ByteBuffer, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(BaseEncoding.base16().encode(ByteBuffers.toByteArray(((ByteBuffer) defaultValue))));
+        break;
+      case DECIMAL:
+        Preconditions.checkArgument(
+            defaultValue instanceof BigDecimal &&
+                ((BigDecimal) defaultValue).scale() == ((Types.DecimalType) type).scale(),
+            "Invalid default %s value: %s",
+            type,
+            defaultValue);
+        generator.writeNumber((BigDecimal) defaultValue);
+        break;
+      case LIST:
+        Preconditions.checkArgument(defaultValue instanceof List, "Invalid default %s value: %s", type, defaultValue);
+        List<Object> defaultList = (List<Object>) defaultValue;
+        Type elementType = type.asListType().elementType();
+        generator.writeStartArray();
+        for (Object element : defaultList) {
+          toJson(elementType, element, generator);
+        }
+        generator.writeEndArray();
+        break;
+      case MAP:
+        Preconditions.checkArgument(defaultValue instanceof Map, "Invalid default %s value: %s", type, defaultValue);
+        Map<Object, Object> defaultMap = (Map<Object, Object>) defaultValue;
+        Type keyType = type.asMapType().keyType();
+        Type valueType = type.asMapType().valueType();
+
+        List<Object> valueList = Lists.newArrayListWithExpectedSize(defaultMap.size());
+        generator.writeStartObject();
+        generator.writeArrayFieldStart(KEYS);
+        for (Map.Entry<Object, Object> entry : defaultMap.entrySet()) {
+          toJson(keyType, entry.getKey(), generator);
+          valueList.add(entry.getValue());
+        }
+        generator.writeEndArray();
+        generator.writeArrayFieldStart(VALUES);
+        for (Object value : valueList) {
+          toJson(valueType, value, generator);
+        }
+        generator.writeEndArray();
+        generator.writeEndObject();
+        break;
+      case STRUCT:
+        Preconditions.checkArgument(
+            defaultValue instanceof StructLike, "Invalid default %s value: %s", type, defaultValue);
+        Types.StructType structType = type.asStructType();
+        List<Types.NestedField> fields = structType.fields();
+        StructLike defaultStruct = (StructLike) defaultValue;
+
+        generator.writeStartObject();
+        for (int i = 0; i < defaultStruct.size(); i++) {
+          Types.NestedField field = fields.get(i);
+          int fieldId = field.fieldId();
+          Object fieldJavaDefaultValue = defaultStruct.get(i, Object.class);
```

Comment:
Nit: this still has "Java" in the variable name.

Reply:
fixed